WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderForm.swift

```diff
@@ -403,7 +403,7 @@ private struct ProductsSection: View {
                 ForEach(viewModel.productRows) { productRow in
                     CollapsibleProductRowCard(viewModel: productRow,
                                               shouldDisableDiscountEditing: viewModel.paymentDataViewModel.isLoading,
-                                              shouldDisallowDiscounts: viewModel.orderHasCoupons,
+                                              shouldDisallowDiscounts: viewModel.shouldDisallowDiscounts,
```

Comment:
Ideally we'll move these properties to their respective view models, so the `CollapsibleProductRowCard` only needs the references to the order and to the product row to be initialized without having to care of the details, but this requires more extensive changes and testing so we've left it outside this PR.

Reply:
Thanks for mentioning this, I have never worked with this screen before and it's interesting to see the approach of having separate viewmodels to handle the complexity of the data it's working with.